From d62db49384611c670aa9c389024d790870df7c6d Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Sun, 01 Sep 2024 14:55:08 +0300
Subject: [PATCH] [FSW-2067] makefile: include pcie_phy_fw.mk root make

in order to generalize the phy fw creation mechanism, the include
to the pcie_phy_fw.mk makefile will be from the root makefile
and it will be located in /arc for all devices

Change-Id: I59f8278085c2be0dc38f4d9c4da88e1047df6f68
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/Makefile b/Makefile
index e455196..e7f4123 100644
--- a/Makefile
+++ b/Makefile
@@ -57,6 +57,7 @@
 PPBOOT_APP_S := $(shell ls app/ppboot/crt0_ppboot.S)
 PPBOOT_APP_S += $(shell ls $(HL_DEV_ARCH)/*.S | grep -v lds)
 
+include $(HL_DEV_ARCH)/pcie_phy_fw.mk
 include $(HL_DEV_ARCH)/Makefile
 
 BINDIR = bin
diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index 684c1cb..c197766 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -1,7 +1,5 @@
 include $(HL_DEV_ARCH_COMMON)/Makefile
 
-include $(HL_DEV_ARCH)/pcie_phy_fw.mk
-
 PPBOOT_MAX_SIZE := 16384
 
 CFLAGS += -I$(SPECS_H15_DIR)/fs1/
@@ -26,6 +24,7 @@
 #	LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 #endif
 
+PHY_FW_SIZE ?= 0	#dummy value for building auxilary apps w/o phy logic
 CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
 
 image_validation: image_validation_common
diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index 51aafdc..1e93451 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -1,10 +1,10 @@
 include $(HL_DEV_ARCH_COMMON)/Makefile
 
-PHY_FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 20480
 
 LDFLAGS += -Wl,--just-symbols $(HL_DEV_ARCH)/exports.elf
 
+PHY_FW_SIZE ?= 0	#dummy value for building auxilary apps w/o phy logic
 CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
 
 EROM_PCIE_DEPS += \
diff --git a/arch/gaudi2/arc/pcie_phy_fw.mk b/arch/gaudi2/arc/pcie_phy_fw.mk
new file mode 100644
index 0000000..2e32023
--- /dev/null
+++ b/arch/gaudi2/arc/pcie_phy_fw.mk
@@ -0,0 +1 @@
+PHY_FW_SIZE := 0x8000
diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index a73c897..8e40add 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -2,7 +2,8 @@
 
 PPBOOT_MAX_SIZE := 18432
 
-include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
+PHY_FW_SIZE ?= 0	#dummy value for building auxilary apps w/o phy logic
+
 CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
 
 CFLAGS += -I $(PREBOOT_ROOT_DIR)/lib/preboot
diff --git a/arch/gaudi3/arc/pcie_phy_fw.mk b/arch/gaudi3/arc/pcie_phy_fw.mk
new file mode 100644
index 0000000..56eb447
--- /dev/null
+++ b/arch/gaudi3/arc/pcie_phy_fw.mk
@@ -0,0 +1,2 @@
+# size must be 16bytes aligned
+PHY_FW_SIZE := 0xA500
\ No newline at end of file
diff --git a/arch/gaudi3/common/pcie_phy_fw.mk b/arch/gaudi3/common/pcie_phy_fw.mk
deleted file mode 100644
index 56eb447..0000000
--- a/arch/gaudi3/common/pcie_phy_fw.mk
+++ /dev/null
@@ -1,2 +0,0 @@
-# size must be 16bytes aligned
-PHY_FW_SIZE := 0xA500
\ No newline at end of file
